Public/Src/Engine/Dll/EngineSchedule.cs

```diff
@@ -810,8 +810,7 @@ private bool ShouldSerializeOptimizationDataStructurePostExecution()
                     // Everything that is not an output under a shared opaque is considered part of the build. 
                     isPathInBuild: path =>
                         // Scheduler.PipGraph.IsPathInBuild is used for extra safety.
-                        scheduler.PipGraph.IsPathInBuild(AbsolutePath.Create(scheduler.Context.PathTable, path)) ||
-                        !SharedOpaqueOutputHelper.IsSharedOpaqueOutput(path), 
+                        scheduler.PipGraph.IsPathInBuild(AbsolutePath.Create(scheduler.Context.PathTable, path)) || !SharedOpaqueOutputHelper.IsSharedOpaqueOutput(path) || (configuration.Schedule.UnsafeDisableSharedOpaqueEmptyDirectoryScrubbing && Directory.Exists(path) && !Directory.EnumerateFileSystemEntries(path).Any()), 

```

Comment:
>(configuration.Schedule.UnsafeDisableSharedOpaqueEmptyDirectoryScrubbing && Directory.Exists(path) && !Directory.EnumerateFileSystemEntries(path).Any() [](start = 172, length = 151)

suggest to move this into its own method to simplify readability. E.g. 'ShouldRemoveEmptyDirectories'
And add some indenting